File: tools/bench/answers/go_sonnet/s3/08_invoice.go
package main

import "fmt"

func main() {
	type item struct {
		name  string
		qty   float64
		price float64
	}
	items := []item{
		{"widget", 4, 2.5},
		{"gizmo", 2, 7.25},
		{"bolt", 10, 0.1},
	}
	total := 0.0
	maxLine := 0.0
	maxName := ""
	for _, it := range items {
		line := it.qty * it.price
		total += line
		if line > maxLine {
			maxLine = line
			maxName = it.name
		}
	}
	fmt.Println(total)
	fmt.Println(maxName)
}

File: tools/bench/answers/rust_sonnet/s3/08_invoice.rs
fn main() {
    let items = [("widget", 4u32, 2.5f64), ("gizmo", 2u32, 7.25f64), ("bolt", 10u32, 0.1f64)];
    let mut total = 0.0f64;
    let mut max_line = 0.0f64;
    let mut max_name = "";
    for &(name, qty, price) in &items {
        let line = qty as f64 * price;
        total += line;
        if line > max_line {
            max_line = line;
            max_name = name;
        }
    }
    println!("{}", total);
    println!("{}", max_name);
}
